### DISCUSSION TITLE
Explicit type for "data" property in query type

### DISCUSSION THREAD

**Discussion by DaveFz** [dotansimha/graphql-code-generator#9478] [Q&A] [answered]
Hello community,

I'm wondering if it's possible to explicitly generate a type for the "data" property in a query type.

For example:

`export type PersonQuery = {
   __typename?: 'Query',
   person?: (
  {
    address: PersonAddressFragment,
    orders: PersonOrderFragment,
  }
 )
}
`

is it possible, to tell the generator to explicitly generate a type for `person?`, so I can use that type throughout my codebase?

Thank you very much in advance :)

**Comment by n1ru4l** [comment, marked as answer]
You can use GraphQL fragments for generating and composing operation partials.

**Reply by DaveFz** [reply]
Ah, by defining a fragment on "root"-level in the query, it generated a fragment for the query.
Thank you :)
